src/components/settings/playlist_settings/PlaylistSettingsComponent.tsx

```diff
@@ -333,36 +363,55 @@ function PlaylistSettingsComponent(props: { bodyRef: any }) {
 					/>
 				</div>
 				<div className={styles.images}>
-					{subFolders.map((folder: IFolder) => (
-						<Folder
-							folder={folder}
-							onClick={() => setCurrentFolder(folder)}
-							draggedElement={draggedElement}
-							onDroppedImage={() => {
-								setDraggedElement(undefined);
-								setCurrentFolder(currentFolder);
-
-								setTimeout(
-									() =>
-										metaDb
-											.getImages(currentFolder.id)
-											.then(setImages),
-									50
-								);
-							}}
-							key={folder.id}
-						/>
-					))}
-
+					{subFolders.map((folder: IFolder) => {
+						if (searchbarValue.trim() === "") {
+							return (
+								<Folder
+									folder={folder}
+									onClick={() => setCurrentFolder(folder)}
+									draggedElement={draggedElement}
+									onDroppedImage={() => {
+										setDraggedElement(undefined);
+										setCurrentFolder(currentFolder);
+										setTimeout(
+											() =>
+												metaDb
+													.getImages(currentFolder.id)
+													.then(setImages),
+											50
+										);
+									}}
+									key={folder.id}
+								/>
+							);
+						}
+					})}
 					{images.map((image: IImage, index: number) => {
+						if (searchbarValue.trim() === "") {
+							return (
+								<Background
+									selected={currentImageId === image.id}
+									image={image}
+									index={index}
+									setDraggedElement={setDraggedElement}
+									key={index}
+								/>
+							);
+						}
+					})}
+					{results.map((image: IImage, index: number) => {
 						return (
-							<Background
-								selected={currentImageId === image.id}
-								image={image}
-								index={index}
-								setDraggedElement={setDraggedElement}
-								key={index}
-							/>
+							<>
+								{
+									<Background
+										selected={false}
+										image={image}
+										index={index}
+										setDraggedElement={setDraggedElement}
+										key={index}
+									/>
+								}
+							</>
 						);
```

Comment:
Why do you use Fragments (`<></>`) and the curley braces here? I don't think they are necessary. If possible, remove the return statement and change the map to an implicit arrow function with brackets.